### DISCUSSION TITLE
Possible to modify values in a parent x-data, when using Alpine.data?

### DISCUSSION THREAD

**Discussion by Saeven** [alpinejs/alpine#4355] [1. Help]
I've got a jsbin that shows what I'm trying to do here:
https://jsbin.com/saziyib/7/edit?html,console,output

The intention is that when you click "Change A to Hello" or "Change B to World" that it changes the variables in the parent.  I suspect that for some reason (of my doing!) it is just changing a locally scoped variable instead of affecting the parent references (a, and b).

Is it possible to achieve what I'm after?  I need 'a' and 'b' to change within the parent container, when the component created by Alpine.data is actuated.

**Comment by SimoTod** [comment]
Alpine.data is a factory, you are creating 2 different components with their own variables.
You can use a store (look for alpine store) or a common parent component storing the variables

**Comment by ekwoka** [comment]
This is an issue of JS knowledge.

You're reassigning the variable.

Not CHANGING the value the variable points to.

To demo

```js
let a = 'hello'
let b = a
a = 'world'
```

`b` will be `'hello'`

**Reply by Saeven** [reply]
I found a way to achieve what I needed, thanks for the discussion.

https://jsbin.com/zejeqew/10/edit?html,output
